tests/components/reolink/conftest.py

```diff
@@ -65,7 +67,15 @@ def reolink_connect(mock_get_source_ip: None) -> Generator[MagicMock, None, None
         host_mock.session_active = True
         host_mock.timeout = 60
         host_mock.renewtimer.return_value = 600
-        yield host_mock
+        yield (host_mock, host_mock_class)
```

Comment:
Just yield the `host_mock_class` here. Tests that want the return value can use the `reolink_connect` fixture below which should return the mock class return value.

Reply:
Thanks for the suggestion, much cleaner now :)